# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tsa/uucp/uucp-1.06.1-vetargs2.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- uucp-1.06.1/uuxqt.c.vetargs	Sun Aug 20 15:29:59 1995
+++ uucp-1.06.1/uuxqt.c	Mon Jan 28 14:19:48 2002
@@ -928,6 +928,32 @@
 	 don't permit multiple arguments.  */
       for (i = 1; azQargs[i] != NULL; i++)
 	{
+	  if (azQargs[i][0] == '-' && azQargs[i][1] == '-')
+	    {
+		char *zopts = azQargs[i] + 2;
+		char *zeq = strchr(zopts, '=');
+
+		/* The -g, -n, and -s options take an argument.  */
+		if (!(strncmp(zopts, "not", 3)	/* --notify */
+		   && strncmp(zopts, "g", 1)	/* --grade */
+		   && strncmp(zopts, "s", 1))	/* --status */
+		&& zeq == NULL)
+		  {
+		    if (azQargs[i+1] != NULL)
+		      ++i;
+		  }
+
+		/* The -I, -u and -x options are not permitted.  */
+		if (!(strncmp(zopts, "con", 3)	/* --config */
+		   && strncmp(zopts, "us", 2)	/* --user */
+		   && strncmp(zopts, "de", 2)))	/* --debug */
+		  {
+		    if (azQargs[i+1] != NULL && zeq == NULL)
+		      ++i;
+		    azQargs[i] = zbufcpy ("--nouucico");
+		  }
+	    }
+	  else
 	  if (azQargs[i][0] == '-')
 	    {
 	      char *zopts;
